src/java/org/apache/ivy/tools/analyser/JarJarDependencyAnalyser.java

```diff
@@ -39,21 +39,22 @@ public JarJarDependencyAnalyser(File jarjarjarLocation) {
     }
 
     public ModuleDescriptor[] analyze(JarModule[] modules) {
-
         StringBuilder jarjarCmd = new StringBuilder("java -jar \"").append(
             jarjarjarLocation.getAbsolutePath()).append("\" --find --level=jar ");
         Map<String, JarModule> jarModulesMap = new HashMap<>();
         Map<ModuleRevisionId, DefaultModuleDescriptor> mds = new HashMap<>();
 
-        for (int i = 0; i < modules.length; i++) {
-            jarModulesMap.put(modules[i].getJar().getAbsolutePath(), modules[i]);
+        for (JarModule jarModule : modules) {
+            jarModulesMap.put(jarModule.getJar().getAbsolutePath(), jarModule);
             DefaultModuleDescriptor md = DefaultModuleDescriptor.newBasicInstance(
-                modules[i].getMrid(), new Date(modules[i].getJar().lastModified()));
-            mds.put(modules[i].getMrid(), md);
-            jarjarCmd.append("\"").append(modules[i].getJar().getAbsolutePath()).append("\"");
-            if (i + 1 < modules.length) {
-                jarjarCmd.append(File.pathSeparator);
-            }
+                    jarModule.getMrid(), new Date(jarModule.getJar().lastModified()));
+            mds.put(jarModule.getMrid(), md);
+            jarjarCmd.append("\"").append(jarModule.getJar().getAbsolutePath()).append("\"");
+            jarjarCmd.append(File.pathSeparator);
+        }
+
+        if (modules.length > 0) {
+            jarjarCmd.setLength(jarjarCmd.length() - 1);
```

Comment:
I don't fully follow what this is block is meant for. Is it to conditionally include the `File.pathSeparator` character?